docs/checkit.js

```diff
@@ -130,6 +130,128 @@ Calendar.prototype.generateMonthObjects = function (startDate, endDate) {
 
 };
 
+// calendarAnalyzer takes a calendar object and can analyze facts about it.
+
+var CalendarAnalyzer = function (calendarState) {
+    "use strict";
+    this.calState = calendarState;
+};
+
+CalendarAnalyzer.prototype.getNumberOfChecked = function () {
+    // Returns the number of checked days in a calendar
+    "use strict";
+    return Object.keys(this.calState.checkedDays).length;
+};
+
+CalendarAnalyzer.prototype.getTotalCalendarDays = function () {
+    // Returns the total number of active days in the calendar
+    "use strict";
+    var endDate = moment(this.calState.endDateString, "YYYYMMDD"),
+        startDate = moment(this.calState.startDateString, "YYYYMMDD"),
+        totalDays = endDate.diff(startDate, 'days');
+    return totalDays;
+};
+
+CalendarAnalyzer.prototype.getNumberOfUnchecked = function (calObj) {
+    // Returns number of unchecked days in a calendar
+    "use strict";
+    return this.getTotalCalendarDays(calObj) - this.getNumberOfChecked(calObj);
+};
+
+CalendarAnalyzer.prototype.getCheckedDaysStreak = function () {
+    // Returns the longest streak of checked days
+    "use strict";
+    // if checkedDays is an empty object, return 0;
+    if (jQuery.isEmptyObject(this.calState.checkedDays)) {
+        return 0;
+    }
+    
+    var currentStreak = [],
+        arrayOfStreakArrays = [],
+        checkedDaysArray = [],
+        key,
+        lastDate,
+        checkedDaysArrayLength,
+        i,
+        currentDay,
+        previousDay,
+        dayDiff,
+        largestArrayIndex;
+    
+    // place checkedDays object's keys into array and sort the array.
+    for (key in this.calState.checkedDays) {
+        if (this.calState.checkedDays.hasOwnProperty(key)) {
+            checkedDaysArray.push(key);
+        }
+    }
+    checkedDaysArray.sort();
+
+    checkedDaysArrayLength = checkedDaysArray.length;
```

Comment:
I was thinking i didn't want to have to run .length every time in the for loop. Does that not matter?